agent/utils/ephemeral_ports.go

```diff
@@ -58,3 +63,82 @@ func GenerateEphemeralPortNumbers(n int, reserved []uint16) ([]uint16, error) {
 	}
 	return result, nil
 }
+
+var dynamicHostPortRange = getDynamicHostPortRange
+
+// GetHostPortRange gets a set of contiguous host ports from the ephemeral host port range defined on the host. The
+// number of host ports is determined by the number of container ports, in the given containerPortRange.
+func GetHostPortRange(containerPortRange string, protocol string) (string, error) {
```

Comment:
Instead of passing in `containerPortRange`, it feels a little more clear to me to simply provide the number of ports needed (length of range):
1. Generalization - this helper method does not need to know what's the container port range (it doesn't even need to know that it's doing it for container port mapping), all it does is to allocate N contiguous host ports. 
2. Redundancy of validation - if we pass in `containerPortRange`, then we need to verify its validity, which, I believe, should be already done during task payload unmarshalling (because we may not be always calling this method, once we support customizable host port range).

Reply:
made the change